src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -554,6 +551,114 @@ def delete_seer_grouping_records(
         Group.objects.bulk_update(groups_with_seer_metadata, ["data"])
 
 
+def single_query_event_per_group(
+    high_event_count_group_ids: list[int], project_id: int
+) -> list[EventGroupSnubaResult]:
+    """
+    Query events for a group one at a time, running the requests in groups of SNUBA_QUERY_RATELIMIT.
+    """
+    group_event_rows: list[EventGroupSnubaResult] = []
+    events_entity = Entity("events", alias="events")
+    for group_ids_chunk in chunked(high_event_count_group_ids, SNUBA_QUERY_RATELIMIT):
```

Comment:
in chunks groups of size SNUBA_QUERY_RATELIMIT (4), run the event query per group.
make a single request of the 4 queries, appending the results to a list